### ISSUE TITLE
Ban mutable `Map`s

### ISSUE THREAD

**Issue by danielnixon** [danielnixon/eslint-config-typed-fp#20]
For example, take these lines: https://github.com/oaf-project/oaf-routing/blob/master/src/page-state-memory.ts#L26-L28

```ts
  // eslint-disable-next-line functional/prefer-readonly-type
  const locations = new Array<LocationKey>();
  const pageStateMap = new Map<LocationKey, PageState>();
```

`new Array` is flagged by `functional/prefer-readonly-type` but nothing flags `new Map`.

Either `prefer-readonly-type` should be updated to catch `new Map` (compare to https://github.com/jonaskello/eslint-plugin-functional/issues/150) or we should update this package do it here: https://github.com/danielnixon/eslint-config-typed-fp/blob/master/src/index.ts#L94-L112

```ts
export const readonlyMap = <K, V>(
  values: Iterable<readonly [K, V]>
): ReadonlyMap<K, V> => new Map(values);
```

**Comment by danielnixon** [comment]
Related: https://github.com/jonaskello/eslint-plugin-functional/issues/151
